rest-api/src/main/scala/za/co/absa/enceladus/rest_api/repositories/v3/RunMongoRepositoryV3.scala

```diff
@@ -0,0 +1,134 @@
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package za.co.absa.enceladus.rest_api.repositories.v3
+
+import org.mongodb.scala.MongoDatabase
+import org.mongodb.scala.bson.BsonDocument
+import org.mongodb.scala.bson.conversions.Bson
+import org.mongodb.scala.model.Aggregates._
+import org.mongodb.scala.model.Projections._
+import org.mongodb.scala.model.Sorts._
+import org.mongodb.scala.model.{Filters, _}
+import org.springframework.beans.factory.annotation.Autowired
+import org.springframework.stereotype.Repository
+import za.co.absa.enceladus.rest_api.models.RunSummary
+import za.co.absa.enceladus.rest_api.repositories.RunMongoRepository
+
+import scala.concurrent.Future
+
+
+@Repository
+class RunMongoRepositoryV3 @Autowired()(mongoDb: MongoDatabase) extends RunMongoRepository(mongoDb) {
+
+  /**
+   * Yields Latest-of-each run summaries (grouped by datasetName, datasetVersion).
+   * Optionally filtered by one of `startDate` (>=)|`sparkAppId`(==)|`uniqueId`(==)
+   * The result is ordered by datasetName, datasetVersion (both ascending)
+   *
+   * @param startDate
+   * @param sparkAppId
+   * @param uniqueId
+   * @return
+   */
+  def getRunSummariesLatestOfEach(datasetName: Option[String] = None,
+                                  datasetVersion: Option[Int] = None,
+                                  startDate: Option[String] = None,
+                                  sparkAppId: Option[String] = None,
+                                  uniqueId: Option[String] = None
+                                 ): Future[Seq[RunSummary]] = {
+    val exclusiveOptsFilter: Option[Bson] = (startDate, sparkAppId, uniqueId) match {
+      case (None, None, None) => None
+      case (Some(startDate), None, None) => Some(startDateFromFilter(startDate))
+      case (None, Some(sparkAppId), None) => Some(sparkIdFilter(sparkAppId))
+      case (None, None, Some(uniqueId)) => Some(Filters.eq("uniqueId", uniqueId))
+      case _ => throw new IllegalArgumentException("At most 1 filter of [startDate|sparkAppId|uniqueId] is allowed!")
+    }
+
+    val datasetFilter: Option[Bson] = datasetNameVersionOptFilter(datasetName, datasetVersion)
+
+    val combinedFilter: Bson = (exclusiveOptsFilter, datasetFilter) match {
+      case (None, None) => BsonDocument() // empty filter
+      case (Some(filter1), None) => filter1
+      case (None, Some(filter2)) => filter2
+      case (Some(filter1), Some(filter2)) => Filters.and(filter1, filter2)
+    }
```

Comment:
What about to make this a routine? Used more times. (Here, lines 98-103,...)

Reply:
Done, thanks for the tip.